server/search-index.js

```diff
@@ -0,0 +1,93 @@
+const fs = require("fs");
+const path = require("path");
+const util = require("util");
+
+const fm = require("front-matter");
+
+const {
+  CONTENT_ROOT,
+  CONTENT_TRANSLATED_ROOT,
+  VALID_LOCALES,
+} = require("../content");
+const { SearchIndex } = require("../build");
+
+const promisifed = util.promisify(fs.readFile);
+
+function* walker(root) {
+  const files = fs.readdirSync(root);
+  for (const name of files) {
+    const filepath = path.join(root, name);
+    const isDirectory = fs.statSync(filepath).isDirectory();
+    if (isDirectory) {
+      yield* walker(filepath);
+    } else {
+      yield filepath;
+    }
+  }
+}
+
+function populateSearchIndex(searchIndex, localeLC) {
+  const root = path.join(
+    localeLC === "en-us" ? CONTENT_ROOT : CONTENT_TRANSLATED_ROOT,
+    localeLC
+  );
+  const filePaths = [...walker(root)].filter((filePath) => {
+    const basename = path.basename(filePath);
+    return basename === "index.html" || basename === "index.md";
+  });
+  const locale = VALID_LOCALES.get(localeLC);
+  return Promise.all(
+    filePaths.map(async (filePath) => {
+      const rawContent = await promisifed(filePath, "utf-8");
+      const { attributes: metadata } = fm(rawContent);
```

Comment:
When having to do ~12k of these, I found it faster to go straight to the `fm()` instead of relying on `Document.read()`. A slight violation of APIs perhaps but totally worth it :)

Reply:
Not the first place we do this. Worth it!